### DISCUSSION TITLE
Why Popover is private

### DISCUSSION THREAD

**Discussion by linasburneika** [iTwin/iTwinUI-react#537] [Q&A]
The Popover component is marked as @private and has to be imported directly from a file to be used. Popover should be reexported at component level. It's usage is already demonstrated in ColorPicker sample, I propose to add additional use case to Data/Time picker component samples with Popover - this is what most users will want to consume directly.

**Comment by veekeys** [comment]
Main reason behind it was that we do not want to support it as our component for possible different user needs. This was intended to be used as a util for our components, not as separate deliverable, where users could use, fill bugs/ask features.
But I do agree exporting it in barrel might be helpful especially with itwin.js coming in...

**Reply by mayank99** [reply]
Every app should have their own Popover though. It is just a thin wrapper around Tippy.js which the user can use instead of our own Popover.

Explicitly marking something as public prevents us from making breaking changes to it and adds maintenance burden.

**Reply by linasburneika** [reply]
I need color/date/time pickers with automatic popup menu (similar as Select dropdown) to be used in itwinjs samples. ColorPickerButton is available from @itwin/imodel-components-react. It's a valid option, however I'd prefer all ui to come from itwinui package. 

to be fair, I don't need Popover as an individual component, I wish those pickers to be an easy to use drop-in consumables. I'm also fine with the approach of DIY dump parts as you currently provide, but then you need to expose an official glue for them too. Take a look at the utils from mui framework.

If it's really a maintenance burden, I'll stick with browser's native "input type="date" instead of adding a bunch of unrelated code. It's simple and does a job.

**Reply by mayank99** [reply]
If I wasn't clear, let me try to elaborate: our `Popover` component is a thin wrapper around an _external_ library called [`tippy.js`](https://atomiks.github.io/tippyjs/) (technically [`@tippyjs/react`](https://github.com/atomiks/tippyjs-react) but it's the same idea). We apply some modifications and opinionated defaults which we think make sense for our deliverable components (like Select and Tooltip). You can use it directly from the file import but it _might not_ work perfectly for your app, and we don't want to be responsible for that. This is why we just recommend using `tippy.js` directly instead of through our wrapper.

However, I agree that we should expose a nicer way of consuming color/date/time pickers. We even have it in our backlog (https://github.com/iTwin/iTwinUI/issues/272) but we don't have an ETA to give you at this time.

I wouldn't recommend using the browser controls as they are not accessible and don't follow our UX standards.

**Reply by linasburneika** [reply]
Thanks, I wasn't aware it's already in the backlog. I need out of the box solution for the pickers, to replace existing simple native controls in the itwinjs samples with components that have consistent look and feel.
